lib/shoryuken/middleware/server/auto_delete.rb

```diff
@@ -2,14 +2,18 @@ module Shoryuken
   module Middleware
     module Server
       class AutoDelete
+        MAX_BATCH_SIZE = 10
+
         def call(worker, queue, sqs_msg, _body)
           yield
 
           return unless worker.class.auto_delete?
 
-          entries = [sqs_msg].flatten.map.with_index { |message, i| { id: i.to_s, receipt_handle: message.receipt_handle } }
+          [sqs_msg].flatten.in_groups_of(MAX_BATCH_SIZE, false).each do |sqs_msgs|
+            entries = sqs_msgs.map.with_index { |message, i| { id: i.to_s, receipt_handle: message.receipt_handle } }
 
-          Shoryuken::Client.queues(queue).delete_messages(entries: entries)
+            Shoryuken::Client.queues(queue).delete_messages(entries: entries)
+          end
```

Comment:
This can also cause duplicate processing depending on the volume. If it takes too long to delete a large batch, messages can go back and be consumed again.

Reply:
That's true, but isn't this true also for any message processing? Even if you're using a regular worker, if it takes too long to process the messages the visibility timeout can expire. I think that the visibility timeout must be carefully chosen in any case.